datura/tools/twitter/twitter_advanced_search_tool.py

```diff
@@ -0,0 +1,120 @@
+import json
+import time
+from typing import Type
+import bittensor as bt
+from neurons.validators.apify.twitter_scraper_actor import TwitterScraperActor
+from pydantic import BaseModel, Field
+from starlette.types import Send
+from datura.tools.base import BaseTool
+from datura.services.twitter_prompt_analyzer import TwitterPromptAnalyzer
+from datura.dataset.date_filters import get_specified_date_filter, DateFilterType
+
+
+class TwitterAdvancedSearchToolSchema(BaseModel):
+    query: str = Field(
+        ...,
+        description="The search query for tweets.",
+    )
+
+
+class TwitterAdvancedSearchTool(BaseTool):
+    """Tool that gets tweets from Twitter."""
+
+    name = "Twitter Advanced Search"
+
+    slug = "get_tweets"
```

Comment:
The slug and tool ID is the same as on the twitter_search_tool. Is this just not finished yet? Both tools are exposed throught the toolkit.